### PR TITLE
Flush() should also clear reference count

### PR DESCRIPTION
**Pull request by xw-g**
* A quick fix to correct the logic that Flush() should also clear reference count.
* Update error message to help debug non zero reference case.
* Update testcase to cover the changes.

### CODE REVIEW

File: rib/rib.go [hunk 1]
```diff
@@ -1725,6 +1732,7 @@ func (r *RIBHolder) Flush() error {
 			backupNHGs = append(backupNHGs, *nhg.BackupNextHopGroup)
 		}
 	}
+	// todo: xw-g check how backup NHG reference is counted.
```
Comment (robshakir) [inline rib/rib.go:1735]:
```suggestion
	// TODO(xw-g): check how backup NHG reference is counted.
```

I'm not sure that we have handling for backup NHG at all right now.

File: rib/rib_test.go [hunk 2]
```diff
@@ -3540,7 +3544,9 @@ func TestFlush(t *testing.T) {
 
 			return r
 		}(),
-	}}
+	},
+	// TODO: xw-g add a test case to test non-zero reference.
```
Comment (robshakir) [inline rib/rib_test.go:3548]: Suggest we should probably do this before we submit the change here? But how is this different from `cannot delete referenced NH` above?

**Comment by robshakir** [comment]
Closing this as obsolete after #221.
